Sources/TuistPlugin/PluginService.swift

```diff
@@ -0,0 +1,89 @@
+import TSCBasic
+import TuistCore
+import TuistLoader
+import TuistSupport
+
+/// A default implementation of `PluginServicing` which loads `Plugins` using
+/// the `Config` description without it needing to be loaded. The  plugins are first fetched/copied into
+/// a cache and the loaded `Plugins` model is returned.
+public final class PluginService: PluginServicing {
```

Comment:
To be consistent with the other files in the codebase and prevent back-and-forths between files, I'd recommend keeping `PluginService`, `PluginServicing`, and `PluginServiceError` in the same file.